grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy

```diff
@@ -56,69 +61,50 @@ class StatusEventService {
+                        durationTime = event.created.time - first.created.time
                     } else if (i < statusEvents.size() - 1) {
                         Event previousEvent = statusEvents.get(i + 1)
-                        duration = new Period(event.created.time, previousEvent.created.time, PeriodType.yearMonthDay())
+                        durationTime = previousEvent.created.time - event.created.time
                     } else {
-                        duration = new Period(event.created.time, new Date().time, PeriodType.yearMonthDay())
+                        durationTime = new Date().time - event.created.time
                     }
-                    new StatusEventDTO(event, duration)
+                    new StatusEventDTO(event, durationTime)
                 }
         eventDTOs
     }
 
-    class StatusEventDTO {
-        Event event
-        Period duration
-        Integer stage
+    List <StatusEventDTO> getStatusEventsForProjectList(List<Issue> issues) {
+        final session = sessionFactory.currentSession
+        final String query = ' select * from event e where e.project_key in :projectKeys ' +
+                             ' and e.event_type in :typeList '
+        final sqlQuery = session.createSQLQuery(query)
 
-        StatusEventDTO(Event event, Period duration) {
-            this.event = event
-            this.duration = duration
-            this.stage = getEventStage(event)
+        List<Event> eventList = sqlQuery.with {
+            addEntity(Event)
+            setParameterList('projectKeys', issues*.projectKey.toList())
+            setParameterList('typeList', TYPES*.name())
+            list()
         }
 
-        private Integer getEventStage(Event event) {
-            switch (event.eventType) {
-                case EventType.INITIAL_STATE:
-                    1 // All project types
-                    break
-                case EventType.GETTING_SIGNATURES:
-                    2 // IRB
-                    break
-                case EventType.REVIEWING_FORM:
-                    2 // Non-IRB
-                    break
-                case EventType.IRB_CONSIDERING:
-                    3 // IRB
-                    break
-                case EventType.GETTING_CCO_APPROVAL:
-                    3 // Non-IRB
-                    break
-                case EventType.APPROVED:
-                    4 // IRB
-                    break
-                case EventType.COMPLETED:
-                    4 // Non-IRB
-                    break
-                case EventType.CLOSED:
-                    5 // IRB
-                    break
-                case EventType.ABANDON:
-                    6
-                    break
-                default:
-                    0
-                    break
+        Map<String, List<Event>> eventMap = eventList.groupBy {it.projectKey}
+        List<StatusEventDTO> result = new ArrayList<>()
+        issues.collect { issue ->
```

Comment:
This returns a collection that you're not using. Probably better to just use `each` in this case.

Reply:
Changed to `each`.